### PR TITLE
docs(libflux): document `fluxcore::scanner` module

### PR DESCRIPTION
**Pull request by pierwill**
Documents all public items in the module, and removes the module-level `#![allow(missing_docs)]` attribute.

Have chosen here not to add a doc comment for each variant of `fluxcore::scanner::TokenType`.

### CODE REVIEW

File: libflux/flux-core/src/scanner/mod.rs [hunk 1]
```diff
@@ -34,31 +35,44 @@ pub struct Scanner {
...
 #[display(fmt = "{}", lit)]
 pub struct Token {
+    /// Type of token.
     pub tok: TokenType,
+    /// String representation of token.
     pub lit: String,
+    /// Offset
     pub start_offset: u32,
+    /// Offset
     pub end_offset: u32,
```
Comment (pierwill) [inline libflux/flux-core/src/scanner/mod.rs:65]: What exactly are these offsets? Characters from the start of a file?

File: libflux/flux-core/src/scanner/mod.rs [hunk 2]
```diff
@@ -209,6 +223,7 @@ impl Scanner {
         }
     }
 
+    /// Produces the next comment token from the input.
```
Comment (pierwill) [inline libflux/flux-core/src/scanner/mod.rs:226]: Wasn't sure about this one.

Comment (fchikwekwe) [reply libflux/flux-core/src/scanner/mod.rs:226]: This sounds right to me.

File: libflux/flux-core/src/scanner/mod.rs (hunk 1 above)
Comment (fchikwekwe) [reply libflux/flux-core/src/scanner/mod.rs:65]: Yes. Its the start and end of the current token within the context of the file.

Comment (fchikwekwe) [reply libflux/flux-core/src/scanner/mod.rs:65]: "Starting location of token" and "ending location of token" or something slightly better worded would probably be more accurate.

**Review by fchikwekwe — COMMENTED** [review]
If you could change the offset comments, that would be great. Otherwise, this looks good. Thanks for helping document this file!

File: libflux/flux-core/src/scanner/mod.rs (hunk 1 above)
Comment (pierwill) [reply libflux/flux-core/src/scanner/mod.rs:65]: Offset in character units, or tokens?

Comment (fchikwekwe) [reply libflux/flux-core/src/scanner/mod.rs:65]: Characters, I'm pretty sure. Not 100% sure if that includes whitespace or not.

Comment (fchikwekwe) [reply libflux/flux-core/src/scanner/mod.rs:65]: I think whitespace is left out. If that matters.

Comment (pierwill) [reply libflux/flux-core/src/scanner/mod.rs:65]: Ah! Definitely characters. :) Always look at the tests, I guess.

https://github.com/influxdata/flux/blob/8d902fdf517445a31af22065541a30584bcc49ef/libflux/flux-core/src/scanner/tests.rs#L4-L18

**Review by fchikwekwe — APPROVED** [review]

**Comment by fchikwekwe** [comment]
Looks good! 🙌🏾

**Pull request merged**